### PR TITLE
Feature/development phase 1

### PR DESCRIPTION
**Pull request by skyv26**
# Achievement `#1` : Todo List Structure

From few days I have learnt a lot about webpack and I really liked it. It really helps us to make the things more dynamic and push the boundaries of `JavaScript`. I have implemented following things as per requirement.

- Setup the repo with `webpack` and `npm-package`.
- Added the template `index.html` file inside `src` folder.
- Added the **list** `placeholders` inside the `index.html` to hold the todo-task.
- Added the `index.js` and created the array list of `object of todos` and render the array list into `index.html` using `HTML Webpack Plugin`.
- Added the `index.scss` file for styles and render the style with the help of `scss-loader`, `css-loader` and `style-loader`.
- Todo list rendered according to index value.
- Finally, Files generated with `webpack` using `npm run build` and served from `dist/` folder using `webpack-dev-serve` command.
 - Added the `linters` and resolve the linters error for `html`, `css/scss` and `javascript`.
 - Added the `github-flow` for quality code check.
 - Used the best practices.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,63 @@
...
+- use git clone in order to get the copy this repo. You need to have atleast basic understanding of JavaScript.
+
+### Setup
+
+- You can simply open it on your vscode in order to start working on it
+
+### Install
+
+- please first install the node package files
+
```
Comment (codecaiine) [inline README.md]: - [x] Good job with your readme. But please add more instructions in order to help non-technical people to run your project locally :smile: 
```
git clone https://github.com/skyv26/todo-list-v1.git

cd todo-list-v1

npm install

npm run build 

npm start
```

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        <i class="fontawesome-icon fa-solid fa-ellipsis-vertical ellipse" id="todo__mode-${each.index}"></i>
+      </div>
+    </li>
+  `;
+  listContainer.insertAdjacentHTML('beforeend', htmlTemplateLiteral);
+});
+
+refreshButton.addEventListener('click', () => {
+  window.location.reload();
+});
```
Comment (codecaiine) [inline src/index.js]: - [x] Good job so far your project is well implemented :+1:  But from the project requirement, the list should appear in order of the index values for each task but it is not the case in your project when I change the index value  :smile: Please fix it . :+1: 
Screenshot from 2022-08-10 20-17-35

**Review by codecaiine — CHANGES_REQUESTED** [review]
Hi @skyv26 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- Great project implementation :heavy_check_mark: 
- No linters errors :heavy_check_mark: 
- Good use of github flow :heavy_check_mark: 
- Descriptive PR :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 2 above)
Comment (skyv26) [reply src/index.js]: @codecaiine Thank you so much. I will update it ASAP.

**Review by codecaiine — APPROVED** [review]
Hi @skyv26 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Great project implementation :heavy_check_mark: 
- No linters errors :heavy_check_mark: 
- Good use of github flow :heavy_check_mark: 
- Descriptive PR :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by skyv26** [comment]
@codecaiine Thank You So Much ❤️

**Pull request merged**